maro/rl/training/async_tools/actor.py

```diff
@@ -0,0 +1,147 @@
+        num_episodes (int): Number of training episodes. Each training episode may contain one or more
+            collect-update cycles, depending on how the implementation of the roll-out manager.
+        num_steps (int): Number of environment steps to roll out in each call to ``collect``. Defaults to -1, in which
+            case the roll-out will be executed until the end of the environment.
+        create_env_wrapper_func (Callable): Function to create an environment wrapper for evaluation. The function
+            should take no parameters and return an environment wrapper instance. If this is None, the training
+            environment wrapper will be used for evaluation. Defaults to None.
+        eval_schedule (Union[int, List[int]]): Evaluation schedule. If an integer is provided, the policies will
+            will be evaluated every ``eval_schedule`` episodes. If a list is provided, the policies will be evaluated
+            at the end of the training episodes given in the list. In any case, the policies will be evaluated
+            at the end of the last training episode. Defaults to None, in which case the policies will only be
+            evaluated after the last training episode.
+        log_env_summary (bool): If True, the ``summary`` property of the environment wrapper will be logged at the end
+            of each episode. Defaults to True.
+        proxy_kwargs: Keyword parameters for the internal ``Proxy`` instance. See ``Proxy`` class
+            for details. Defaults to the empty dictionary.
+        log_dir (str): Directory to store logs in. A ``Logger`` with tag "LOCAL_ROLLOUT_MANAGER" will be created at init
+            time and this directory will be used to save the log files generated by it. Defaults to the current working
+            directory.
+    """
+    if num_steps == 0 or num_steps < -1:
+        raise ValueError("num_steps must be a positive integer or -1")
+
+    env_wrapper = create_env_wrapper_func()
+    eval_env_wrapper = env_wrapper if not create_eval_env_wrapper_func else create_eval_env_wrapper_func()
+    agent_wrapper = create_agent_wrapper_func()
+    peers = {"policy_server": 1}
+    proxy = Proxy(group, "actor", peers, **proxy_kwargs)
+    policy_server_address = proxy.peers["policy_server"][0]
+    logger = Logger(proxy.name, dump_folder=log_dir)
+    policy_version = None
+
+    # evaluation schedule
+    if eval_schedule is None:
+        eval_schedule = []
+    elif isinstance(eval_schedule, int):
+        num_eval_schedule = num_episodes // eval_schedule
+        eval_schedule = [eval_schedule * i for i in range(1, num_eval_schedule + 1)]
+
+    eval_schedule.sort()
+    if not eval_schedule or num_episodes != eval_schedule[-1]:
+        eval_schedule.append(num_episodes)
+    eval_point_index = 0
+
+    # get initial policy states from the policy manager
+    if isinstance(agent_wrapper, AgentWrapper):
```

Comment:
what if not?

replace it by an assertion?

Reply:
I had remote inference in this branch before where the agent_wrapper would've been an instance of ``PolicyClient`` and it would not have been necessary to pull policies from the server. Since we are no longer dealing with remote inference in this PR, I will remove this if-statement.